shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/ProxyBackendHandlerFactory.java

```diff
@@ -106,16 +107,14 @@ public static ProxyBackendHandler newInstance(final DatabaseType databaseType, f
         databaseType.handleRollbackOnly(connectionSession.getTransactionStatus().isRollbackOnly(), sqlStatement);
         checkUnsupportedSQLStatement(sqlStatement);
         if (sqlStatement instanceof DistSQLStatement) {
-            if (connectionSession.getTransactionStatus().isInTransaction() && !(sqlStatement instanceof RQLStatement || sqlStatement instanceof QueryableRALStatement)) {
-                throw new UnsupportedOperationException("Non-query dist sql is not supported within a transaction");
-            }
+            checkUnsupportedDistSQLStatementInTransaction(sqlStatement, connectionSession);
             return DistSQLBackendHandlerFactory.newInstance((DistSQLStatement) sqlStatement, connectionSession);
         }
         SQLStatementContext<?> sqlStatementContext = SQLStatementContextFactory.newInstance(ProxyContext.getInstance().getContextManager().getMetaDataContexts().getMetaData().getDatabases(),
                 sqlStatement, connectionSession.getDefaultDatabaseName());
         return newInstance(databaseType, new LogicSQL(sqlStatementContext, sql, Collections.emptyList()), connectionSession, false);
     }
-    
+
```

Comment:
Please keep the indent consistent with the previous line.

Reply:
ok

Reply:
<img width="900" alt="image" src="https://user-images.githubusercontent.com/19788130/182631921-747c8a97-44e3-40a1-9bb8-59cc70a40590.png">

The content of this line is the same as before, the modification of this line you see comes from my first commit.